uvicorn/supervisors/basereload.py

```diff
@@ -22,22 +22,37 @@ def __init__(self, config, target, sockets):
         self.sockets = sockets
         self.should_exit = threading.Event()
         self.pid = os.getpid()
+        self.child_pids = []
 
     def signal_handler(self, sig, frame):
         """
         A signal handler that is registered with the parent process.
         """
+        logger.info(f"Handling signal: {sig}")
+        for child_pid in self.child_pids:
+            logger.debug(f"Attempt at killing child PID {child_pid}")
+            try:
+                os.kill(child_pid, signal.SIGINT)
+                (pid, status) = os.waitpid(child_pid, 0)
```

Comment:
Any thoughts on my suggestion to add a reusable util in `uvicorn/subprocess.py`? My rationale is that this is fairly "sensitive" logic, it shouldn't be duplicated.

```python
def stop_subprocess(pid):
    os.kill(pid, signal.SIGINT)
    os.waitpid(pid, 0)
```

Reply:
yep did it !